pyquil/api/_qam.py

```diff
@@ -60,18 +61,24 @@ def load(self, executable):
         return self
 
     @_record_call
-    def write_memory(self, *, region_name: str, offset: int = 0, value=None):
+    def write_memory(self, *, region_name: str, offset: int = 0, value: Optional[Union[int, float, Sequence[int], Sequence[float]]] = None):
```

Comment:
I think that we should change `offset` so that it defaults to `None` and warn people if they pass a `Sequence` for a `value` and `offset` is not `None`. In addition, line is too long. Reach out if you have any questions.

Reply:
Is there a reason we're assuming the offset should be None when passing a Sequence?